### PR TITLE
Stimmy

### PR DESCRIPTION
**Pull request by derHowie**
BX-1501
BX-1494
BX-1493
BX-1488
BX-1485
RNBW-4768
BX-1526

Figma link (if any): https://www.figma.com/design/p9GJErqWsuSke4Xr2M7et6/eth-rewards?node-id=213-10585&t=vu9PALi7qvAF7ErO-0

## What changed (plus any additional context for devs)
Added stimmy UI + stim flow. Adding `INTERNAL_BUILD=true` will trigger the experiment. When `IS_TESTING=true` the UI will use mock data and will not actually fire off the claim event. Designs linked are for mobile so you will see some slight differences. Optimism is currently the only selectable network.

## Screen recordings / screenshots
See loom videos linked in BX channel if you need to reference the intended flow.

## What to test
I would start with `IS_TESTING=true` so that you can go through the flow as many times as you want. When you actually claim, you'll need to contact BE and ask for a rotation to do it again on your test accounts. Navigate to the points screen. Check out the new tabs, ensure the new UI is displaying correctly. Proceed through the claim flow, ensure that the points UI updates to reflect your claim (this will not happen with mock data because the claim will always appear available)

### CODE REVIEW

File: src/core/raps/unlockAndCrosschainSwap.ts [hunk 1]
```diff
@@ -135,6 +137,13 @@ export const createUnlockAndCrosschainSwapRap = async (
       assetToSell.address?.toLowerCase() as keyof PermitSupportedTokenList
     ];
 
+  if (claimHash) {
+    const claim = createNewAction('claim', {
+      claimHash: claimHash,
+    });
+    actions = actions.concat(claim);
+  }
```
Comment (estebanmino) [inline src/core/raps/unlockAndCrosschainSwap.ts:145]: maybe i'm lost but why don't we have another rap for this?

needing a claimHash feels like you already send the tx and raps are built to execute the tx

File: src/entries/popup/pages/home/Points/ClaimOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+      return claimInfo;
+    },
+    onSuccess: async (d) => {
+      if (
+        claimNetwork !== ChainId.optimism &&
+        d?.txHash &&
+        opEth &&
+        destinationEth
+      ) {
```
Comment (estebanmino) [inline src/entries/popup/pages/home/Points/ClaimOverview.tsx:115]: is there any option that this if happens? for example, if  `claimNetwork !== ChainId.optimism` should we execute the mutationFn ? feels like we should be using this if condition to execute the fn not after is executed (potentially using wrong params)

File: src/entries/popup/pages/home/Points/ClaimOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+  const showPreparingClaim = !showSummary;
+  const showSuccess =
+    !waitToDisplay && claimFinished && !showSummary && !error && !bridgeError;
+  const showError = !waitToDisplay && (error || bridgeError);
+
+  useEffect(() => {
+    claimRewards();
+    setTimeout(() => setWaitToDisplay(false), 3000);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (estebanmino) [inline src/entries/popup/pages/home/Points/ClaimOverview.tsx:175]: will this execute twice if you reload the extension? considering this is a routed path         path: ROUTES.CLAIM_OVERVIEW,

File: src/entries/popup/pages/home/Points/ClaimSheet.tsx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        paddingVertical="10px"
+        className={rowTransparentAccentHighlight}
+        borderRadius="12px"
+        as={motion.div}
+        whileTap={{ scale: 0.98 }}
+        whileFocus={{ scale: 1.02 }}
+        whileHover={{ scale: 1.02 }}
+        paddingHorizontal="8px"
+        onClick={() => {
+          if (chain !== ChainId.optimism) return;
```
Comment (estebanmino) [inline src/entries/popup/pages/home/Points/ClaimSheet.tsx:148]: i see base and zora as chains to be used here, click won't do anything for those cases?

File: static/json/languages/en_US.json [hunk 5]
```diff
@@ -1452,10 +1457,37 @@
...
-
       "your_earnings": "Your Earnings Last Week",
       "view_breakdown": "View your weekly earnings breakdown",
       "view_now": "View Now"
+    },
+    "rewards": {
+      "counting_points": "> Counting earned points...",
+      "calculating_referrals": "> Calculating your referrals...",
+      "allocating_earnings": "> Allocating your earnings...",
+      "claiming_your_reward": "\u00A0\u00A0\u00A0\u00A0\u00A0\u00A0\u00A0\u00A0CLAIMING YOUR REWARD",
```
Comment (estebanmino) [inline static/json/languages/en_US.json:1468]: feels like the `\u00A0` should be in the code and not languages json

File: src/entries/popup/pages/home/Points/ClaimOverview.tsx (hunk 2 above)
Comment (derHowie) [reply src/entries/popup/pages/home/Points/ClaimOverview.tsx:122]: discussed offline, added a clarifying comment

File: src/entries/popup/pages/home/Points/ClaimSheet.tsx (hunk 4 above)
Comment (derHowie) [reply src/entries/popup/pages/home/Points/ClaimSheet.tsx:149]: discussed offline, other networks are currently unavailable

File: src/entries/popup/pages/home/Points/ClaimOverview.tsx (hunk 3 above)
Comment (derHowie) [reply src/entries/popup/pages/home/Points/ClaimOverview.tsx:52]: refactored, claimRewards is now called from the parent on network selection

File: src/core/raps/unlockAndCrosschainSwap.ts (hunk 1 above)
Comment (derHowie) [reply src/core/raps/unlockAndCrosschainSwap.ts]: some random guy stepped in and took care of this

File: src/core/raps/claimAndBridge.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import { createNewAction, createNewRap } from './common';
+import { RapAction, RapClaimActionParameters } from './references';
+
+export const createClaimAndBridgeRap = async (
+  claimParameters: RapClaimActionParameters,
+) => {
+  let actions: RapAction<'crosschainSwap' | 'claim' | 'claimBridge'>[] = [];
```
Comment (jinchung) [inline src/core/raps/claimAndBridge.ts:7]: don't need `crosschainSwap` here

File: src/core/raps/actions/crosschainSwap.ts [hunk 7]
```diff
@@ -64,6 +64,8 @@ export const estimateCrosschainSwapGasLimit = async ({
       );
     }
 
+    console.log('about to call estimateGasWithPadding');
```
Comment (brunobar79) [inline src/core/raps/actions/crosschainSwap.ts:67]:
```suggestion
```

File: src/core/raps/actions/crosschainSwap.ts [hunk 8]
```diff
@@ -74,6 +76,7 @@ export const estimateCrosschainSwapGasLimit = async ({
       provider: provider,
       paddingFactor: SWAP_GAS_PADDING,
     });
+    console.log('estimateGasWithPadding returned', gasLimit);
```
Comment (brunobar79) [inline src/core/raps/actions/crosschainSwap.ts:79]:
```suggestion
```

File: src/entries/popup/pages/home/Points/Points.tsx [hunk 9]
```diff
@@ -280,9 +281,19 @@ export function Points() {
   const { currentAddress } = useCurrentAddressStore();
   const { data, isInitialLoading } = usePoints(currentAddress);
 
+  console.log(
+    'aaaa config,INTERNAL_BUILD ',
+    config,
+    process.env.INTERNAL_BUILD,
+  );
```
Comment (brunobar79) [inline src/entries/popup/pages/home/Points/Points.tsx:288]:
```suggestion
```

**Review by brunobar79 — APPROVED** [review]

**Review by DanielSinclair — APPROVED** [review]

**Pull request merged**